Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -77,149 +78,207 @@ extension NIOClientTCPBootstrap {
-                        // through a proxy we add it later.
-                        let completionPromise = channel.eventLoop.makePromise(of: Void.self)
-                        try channel.pipeline.syncOperations.addHandler(TLSEventsHandler(completionPromise: completionPromise), name: TLSEventsHandler.handlerName)
+        configuration: HTTPClient.Configuration,
+        sslContextCache: SSLContextCache,
+        logger: Logger
+    ) -> EventLoopFuture<NIOClientTCPBootstrap> {
+        return self.makeBootstrap(on: eventLoop,
+                                  host: host,
+                                  requiresTLS: requiresTLS,
+                                  configuration: configuration,
+                                  sslContextCache: sslContextCache,
+                                  logger: logger)
+            .map { bootstrap in
+                bootstrap.channelOption(ChannelOptions.socket(SocketOptionLevel(IPPROTO_TCP), TCP_NODELAY), value: 1)
+                    .channelInitializer { channel in
+                        do {
+                            if let proxy = configuration.proxy {
+                                try channel.pipeline.syncAddProxyHandler(host: host, port: port, authorization: proxy.authorization)
+                            } else if requiresTLS {
+                                // We only add the handshake verifier if we need TLS and we're not going through a proxy. If we're going
+                                // through a proxy we add it later.
+                                let completionPromise = channel.eventLoop.makePromise(of: Void.self)
+                                try channel.pipeline.syncOperations.addHandler(TLSEventsHandler(completionPromise: completionPromise), name: TLSEventsHandler.handlerName)
+                            }
+                            return channel.eventLoop.makeSucceededVoidFuture()
+                        } catch {
+                            return channel.eventLoop.makeFailedFuture(error)
+                        }
                     }
-                    return channel.eventLoop.makeSucceededVoidFuture()
-                } catch {
-                    return channel.eventLoop.makeFailedFuture(error)
-                }
             }
     }
 
-    static func makeHTTP1Channel(destination: ConnectionPool.Key, eventLoop: EventLoop, configuration: HTTPClient.Configuration, preference: HTTPClient.EventLoopPreference) -> EventLoopFuture<Channel> {
+    static func makeHTTP1Channel(destination: ConnectionPool.Key,
+                                 eventLoop: EventLoop,
+                                 configuration: HTTPClient.Configuration,
+                                 sslContextCache: SSLContextCache,
+                                 preference: HTTPClient.EventLoopPreference,
+                                 logger: Logger) -> EventLoopFuture<Channel> {
```

Comment:
Since this method is the only one which is invoked from the outside, do we maybe want to move this to the top of this extension?

Reply:
I can but it'd make the diff larger and much harder to review to change. Up to you and the other reviewers, usually I try to keep the diff small if reasonably possible.

Reply:
I agree with the general goal. However the `Utils.swift` has exploded for me so much, that I have checked out your branch anyway and only come here, to post my comments.

Reply:
it's now at the top.